elastic/kibana
# [Monitoring] Use async/await

Resolves #81104
Resolves #72007
Resolves #81166
Resolves #81084 
Resolves #81225
Resolves #81312
Resolves #53159
Resolves #81340
Resolves #81113

I'm not really sure what's going on here.

I ran tests locally on 7.x and didn't experience any failure. Then, I decided to load up the es archive data and play around with the app to make sure things looked fine. I noticed that I got into an infinite navigation loop by simply loading the cluster overview page and then clicking `Clusters` in the top left. It went from `#/home` to `#/overview` back and forth forever.

I added some debugging and found that [this promise](https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/public/services/clusters.js#L40) was never resolving (successfully or with an error) for some reason when we went from the `#/home` page to the `#/overview`. I decided to try and remove the promises to make it easier to read and reason and that seemed to solve it for me, but I welcome someone else to try and reproduce this.

## Testing

1) Load up 7.x Kibana and ES (without SSL because I don't know how to load archive data with SSL)
2) From the `x-pack` folder, load a sample archive:
`node ../scripts/es_archiver.js load monitoring/singlecluster-three-nodes-shard-relocation --kibana-url http://elastic:changeme@localhost:5601 --es-url http://elastic:changeme@localhost:9200`
3) Navigate to the Stack Monitoring UI and adjust the time picker to a range that includes the archive data (for this example, 10/5/17 -> 10/6/17 works)
4) Allow the cluster overview page to naturally load
5) Click `Clusters` in the top left and observe the infinite navigation loop
